env/production/aws-iam-role-GitHubActionsRoleNextstrainBatchJobs.tf

```diff
@@ -21,7 +21,7 @@ resource "aws_iam_role" "GitHubActionsRoleNextstrainBatchJobs" {
         "Condition": {
           "StringLike": {
             "token.actions.githubusercontent.com:aud": "sts.amazonaws.com",
-            "token.actions.githubusercontent.com:sub": "repo:nextstrain/.github:*"
+            "token.actions.githubusercontent.com:sub": "repo:nextstrain/*:*:job_workflow_ref:nextstrain/.github/.github/workflows/pathogen-repo-build.yaml@refs/heads/*",
```

Comment:
Yes, it would… and that was kinda intentional. I was worried that forks could get a specific commit id accessible in workflow calls. Reason is that GitHub (roughly speaking) stores all forks in the same Git storage backend, which means you can do things like view commits in forks via the forked repo, e.g. https://github.com/nextstrain/test-github-actions/commit/0d1fee0a171c2a8344a8aed8781fd1d32fbd82ec which really comes from https://github.com/tsibley/test-github-actions.

However, GitHub's protected against [such attacks](https://github.com/tsibley/test-github-actions/actions/runs/9116651275):

![image](https://github.com/nextstrain/infra/assets/79913/3ed6bd6d-51f0-4dac-bbe6-f0dc49047efb)

(In hindsight, of course they have, but they've also had other oversights in Actions before, so I don't think I was wrong to be wary.)